.github/workflows/build-publish.yml

```diff
@@ -0,0 +1,181 @@
+name: Build
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+    inputs:
+      force_thunderstore:
+        description: Force re-publishing to Thunderstore
+        required: true
+        default: false
+        type: boolean
+      force_nuget:
+        description: Force re-publishing to NuGet
+        required: true
+        default: false
+        type: boolean
+    
+permissions:
+  contents: write
+  id-token: write
+  attestations: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+      attestation-url: ${{ steps.attest.outputs.attestation-url }}
+      # when you are ready to publish your first release, change this to 'true'
+      allow-release: 'false'
+      is-new-version: ${{ !fromJSON(steps.needs-release.outputs.exists || 'true') }}
+
+    steps:
+    - uses: actions/checkout@v5
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v5
+      with:
+        dotnet-version: 9.x
+    - name: Build
+      run: dotnet build -c Release --verbosity normal
+    - name: Pack
+      run: dotnet pack -o nuget/
+    - name: Get Version
+      id: version
+      run: |-
+        VERSION=$(dotnet build Prepatcher/SilksongPrepatcher.csproj -getProperty:Version)
+        echo Version is $VERSION
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+    - name: Pack Thunderstore
+      run: dotnet tcli build --config-path thunderstore/thunderstore.toml --package-name SilksongPrepatcher --package-version ${{ steps.version.outputs.version }}
```

Comment:
Package name can go into thunderstore.toml if we're not getting it from the build system. Semi-certain it would be 

```toml
[package]
name = "SilksongPrepatcher"
```